app/src/main/res/layout/ingredient_item_fragment.xml

```diff
@@ -3,125 +3,115 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    >
+    android:layout_height="match_parent">
 
 
-    <androidx.fragment.app.FragmentContainerView
```

Comment:
Did you get rid of this for the rounded rectangle background? Because I believe you can still have that with the fragment container. Though I think this is why the layout is janked, since quite a few things use the containerview as a constraint

Reply:
No I got rid of it because the logic for creating the alert dialog fragment - I was getting an error when trying to inflat e the layout with this file

Reply:
Fair enough, I'll try it with the containerview locally, but your code is fine right now though